cloud-functions/src/index.ts

```diff
@@ -2825,3 +2827,49 @@ export const getSharedAddressBook = functions.https.onRequest(async (req, res) =
 		}
 	});
 });
+
+export const getMultisigHistory = functions.https.onRequest(async (req, res) => {
+	corsHandler(req, res, async () => {
+		const signature = req.get('x-signature');
+		const address = req.get('x-address');
+		const network = String(req.get('x-network'));
+
+		const { isValid, error } = await isValidRequest(address, signature, network);
+		if (!isValid) return res.status(400).json({ error });
+
+		const { multisigAddress, limit, page } = req.body;
+		if (!multisigAddress || !network || isNaN(limit) || isNaN(page)) return res.status(400).json({ error: responseMessages.missing_params });
+		if (Number(limit) > 100 || Number(limit) <= 0) return res.status(400).json({ error: responseMessages.invalid_limit });
+		if (Number(page) <= 0) return res.status(400).json({ error: responseMessages.invalid_page });
+
+		try {
+			const encodedMultisigAddress = encodeAddress(multisigAddress, chainProperties[network].ss58Format);
+			const { data: historyItemsArr, error: historyItemsError, count } = await getHistoryTransactions(
+				encodedMultisigAddress,
+				network,
+				Number(limit),
+				Number(page),
+				firestoreDB
+			);
+
+			if (historyItemsError || !historyItemsArr) return res.status(400).json({ error: historyItemsError || responseMessages.queue_fetch_error });
+
+			res.status(200).json({ data: historyItemsArr, count });
```

Comment:
count should be returned inside data for consistency.
data and error should be the only things returned from the cloud functions